### PR TITLE
Add hover update check

### PR DESCRIPTION
**Pull request by jld3103**

### CODE REVIEW

File: internal/versioncheck/version.go [hunk 1]
```diff
@@ -9,10 +9,11 @@ import (
...
-	"github.com/go-flutter-desktop/hover/internal/fileutils"
-	"github.com/go-flutter-desktop/hover/internal/log"
 	"github.com/pkg/errors"
 	"github.com/tcnksm/go-latest"
+
+	"github.com/go-flutter-desktop/hover/internal/fileutils"
+	"github.com/go-flutter-desktop/hover/internal/log"
 )
 
 // CheckForGoFlutterUpdate check the last 'go-flutter' timestamp we have cached
```
Comment (pchampio) [inline internal/versioncheck/version.go:19]: maybe we can factorize both fonctions?

Comment (jld3103) [reply internal/versioncheck/version.go:19]: Ok

**Comment by jld3103** [comment]
@pchampio I think this is much better now

File: cmd/bumpversion.go [hunk 2]
```diff
@@ -43,7 +43,6 @@ func upgrade(targetOS string) (err error) {
 	} else {
 		engineCachePath = enginecache.ValidateOrUpdateEngine(targetOS, "")
 	}
-	engineCachePath = enginecache.ValidateOrUpdateEngine(targetOS, buildCachePath)
```
Comment (GeertJohan) [inline cmd/bumpversion.go:46]: Is the removal of this line related to version update/check?

Comment (jld3103) [reply cmd/bumpversion.go:46]: Nope, but I found it while going through the code. Just some random leftover I think.

**Pull request merged**